sdk/python/foundation-models/cohere/cohere-aisearch-rag.ipynb

```diff
@@ -0,0 +1,483 @@
+    "    },\n",
+    "    {\n",
+    "        \"hotelId\": \"5\",\n",
+    "        \"hotelName\": \"Secret Point\",\n",
+    "        \"description\": \"One of the best hotel in town. The hotel is ideally located on the main commercial artery of the city in the heart of New York.\",\n",
+    "        \"category\": \"Boutique\",\n",
+    "    },\n",
+    "]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create the embeddings\n",
+    "We will create a vector field for each of the hotel descriptions"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# embed each of the descriptions\n",
+    "# you will notice that Cohere has a field called \"input_type\" which can be set to \"search_document\", \"search_query\", \"classification\", or \"clustering\" depedning on the text you are embedding\n",
+    "for doc in docs_to_index:\n",
+    "    doc[\"descriptionVector\"] = co_embed.embed(texts=[doc[\"description\"]], input_type='search_document').embeddings[0]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create the index\n",
+    "Next we will create the index using the Azure SDK"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# create the search index in azure\n",
+    "index_client = SearchIndexClient(\n",
+    "    endpoint=search_service_endpoint, credential=key_credential)\n",
+    "fields = [\n",
+    "    SimpleField(name=\"hotelId\", type=SearchFieldDataType.String, key=True, sortable=True, filterable=True, facetable=True),\n",
+    "    SearchableField(name=\"hotelName\", type=SearchFieldDataType.String),\n",
+    "    SearchableField(name=\"description\", type=SearchFieldDataType.String),\n",
+    "    SearchableField(name=\"category\", type=SearchFieldDataType.String,\n",
+    "                    filterable=True),\n",
+    "    SearchField(name=\"descriptionVector\", type=SearchFieldDataType.Collection(SearchFieldDataType.Single),\n",
+    "                searchable=True, vector_search_dimensions=1024, vector_search_profile_name=\"myHnswProfile\")\n",
+    "]\n",
+    "\n",
+    "# Configure the vector search configuration  \n",
+    "vector_search = VectorSearch(\n",
+    "    algorithms=[\n",
+    "        HnswAlgorithmConfiguration(\n",
+    "            name=\"myHnsw\"\n",
+    "        )\n",
+    "    ],\n",
+    "    profiles=[\n",
+    "        VectorSearchProfile(\n",
+    "            name=\"myHnswProfile\",\n",
+    "            algorithm_configuration_name=\"myHnsw\",\n",
+    "        )\n",
+    "    ]\n",
+    ")\n",
+    "\n",
+    "# Create the search index with the semantic settings\n",
+    "index = SearchIndex(name=index_name, fields=fields,\n",
+    "                    vector_search=vector_search)\n",
+    "result = index_client.create_or_update_index(index)\n",
```

Comment:
Is the key wrapped with AzureKeyCredential in the constants cell? SHould be:
key_credential = AzureKeyCredential("AZURE_SEARCH_ADMIN_KEY")

Reply:
I think you are setting the `key_credential` to a string instead of a `AzureKeyCredential` object as the code in the second cell suggests like this:

```
key_credential = AzureKeyCredential("AZURE_SEARCH_ADMIN_KEY")
```

Reply:
@stewart-co sorry missed your comment. Yeah, that's the problem.

@balapv You probably faced this error in another notebook `cohere-aisearch-langchain-rag.ipynb` which I fixed in the latest commit.